main.js

```diff
@@ -1,60 +1,107 @@
-let redirect = function(url) {
-	window.location.href = url;
-}
-
-let redirectCooldown = function(self, url, cooldown) {
-	if (--cooldown.innerText <= 0)
-	{
-		clearInterval(self[0]);
-		redirect(url);
-	}
-}
-
-window.onload = function() {
-	let moduleName = document.getElementById("moduleName");
-	let moduleIcon = document.getElementById("moduleIcon");
-	let cooldownText = document.getElementById("cooldownText");
-	let redirectButton = document.getElementById("redirectButton");
-
-	try {
-		let moduleData = document.location.search.match(/[?&]redirect=(\w+)/i)[1].toLocaleLowerCase();
-		if (!(data.hasOwnProperty(moduleData))) throw null;
-
-		document.title = "Redirecting to #" + moduleData;
-
-		moduleName.innerText = moduleData;
-		moduleData = data[moduleData];
-		document.getElementById("owner").innerText = moduleData.owner;
-		if (moduleData.hasOwnProperty("icon"))
-			moduleIcon.src = `https://i.imgur.com/${moduleData.icon}.png`;
-
-		if (document.location.search.match(/[?&]hasbutton=true/i))
-		{
-			cooldownText.remove();
-			redirectButton.classList.remove("hidden");
-
-			redirectButton.addEventListener("click", () => redirect(moduleData.url), false);
-		}
-		else
-		{
-			redirectButton.remove();
-			cooldownText.classList.remove("hidden");
-
-			let id = [ ];
-			id[0] = setInterval(redirectCooldown, 1000, id, moduleData.url, document.getElementById("cooldown"));
-		}
-	}
-	catch(_)
-	{
-		document.title = "Invalid module.";
-
-		moduleName.classList.remove("yellow");
-		moduleName.classList.add("red");
-
-		document.getElementById("moduleOwner").innerText = "Invalid module.";
-		moduleIcon.src = "https://i.imgur.com/v4Iurtp.png";
-
-		cooldownText.remove();
-		redirectButton.remove();
-	}
-}
\ No newline at end of file
+const BASE_URI = "https://a801-luadev.github.io"
+
+let redirect = function(url) {
+	window.location.href = url;
+}
+
+let redirectCooldown = function(self, url, cooldown) {
+	if (--cooldown.innerText <= 0)
+	{
+		clearInterval(self[0]);
+		redirect(url);
+	}
+}
+
+let getMatchedModules = function(query) {
+	let matches = {};
+	for (let name of Object.keys(data)) {
+		if (name.includes(query)) {
+			matches[name] = name.match(new RegExp(query, "gi") || []).length;			
+		}
+	}
+	return Object.entries(matches).sort((e1, e2) => e2[1] - e1[1]);
+}

```

Comment:
Makes sort function external so that it doesn't get recreated

```suggestion
let modulesSort = function(e1, e2) {
	return e2[1] - e1[1]
}

let getMatchedModules = function(query) {
	let matches = {};
	for (let name of Object.keys(data)) {
		if (name.includes(query))
			matches[name] = (name.match(new RegExp(query, "gi")) || []).length;			
	}
	return Object.entries(matches).sort(modulesSort);
}

```